hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/ITestAbfsBlobClient.java

```diff
@@ -0,0 +1,151 @@
+    } catch (AzureBlobFileSystemException ex) {
+      // Todo: [FnsOverBlob] - Remove this block once all Blob Endpoint Support is ready.
+      Assertions.assertThat(ex.getMessage()).contains("Blob Endpoint Support is not yet implemented");
+    }
+  }
+
+  private void testClientAPIs(AbfsClient client, TracingContext tracingContext) throws Exception {
+    // 1. Set File System Properties
+    String val1 = Base64.encode("value1".getBytes());
+    String val2 = Base64.encode("value2".getBytes());
+    String properties = "key1=" + val1 + ",key2=" + val2;
+    client.setFilesystemProperties(properties, tracingContext);
+
+    // 2. Get File System Properties
+    client.getFilesystemProperties(tracingContext);
+
+    // 3. Create Path
+    client.createPath("/test", true, true, null, false, null, null,  tracingContext);
+    client.createPath("/dir", false, true, null, false, null, null,  tracingContext);
+    client.createPath("/dir/test", true, true, null, false, null, null,  tracingContext);
+
+    // 4. List Path
+    client.listPath("/", false, 5, null, tracingContext);
+
+    // 5. Acquire lease
+    client.acquireLease("/dir/test", 5, tracingContext);
+
+    // 6. Set Path Properties
+    client.setPathProperties("/test", properties, tracingContext, null);
+
+    // 7. Get Path Status
+    client.getPathStatus("/test", true, tracingContext, null);
+
+    // N. Delete File System
+    client.deleteFilesystem(tracingContext);
+  }
+
+  private AzureBlobFileSystem getBlobFileSystem() throws Exception {
+    Configuration rawConfig = new Configuration();
+    rawConfig.addResource(TEST_CONFIGURATION_FILE_NAME);
+
+    String fileSystemName = TEST_CONTAINER_PREFIX + UUID.randomUUID().toString();
+    String accountName = rawConfig.get(FS_AZURE_ACCOUNT_NAME, "");
+    if (accountName.isEmpty()) {
+      // check if accountName is set using different config key
+      accountName = rawConfig.get(FS_AZURE_ABFS_ACCOUNT_NAME, "");
+    }
+    Assume.assumeFalse("Skipping test as account name is not provided", accountName.isEmpty());
+
+    Assume.assumeFalse("Blob Endpoint Works only with FNS Accounts",
+        rawConfig.getBoolean(FS_AZURE_TEST_NAMESPACE_ENABLED_ACCOUNT, true));
+    accountName = setBlobEndpoint(accountName);
+
+    AbfsConfiguration abfsConfig = new AbfsConfiguration(rawConfig, accountName);
+    AuthType authType = abfsConfig.getEnum(FS_AZURE_ACCOUNT_AUTH_TYPE_PROPERTY_NAME, AuthType.SharedKey);
+    String abfsScheme = authType == AuthType.SharedKey ? FileSystemUriSchemes.ABFS_SCHEME
+        : FileSystemUriSchemes.ABFS_SECURE_SCHEME;
+    final String abfsUrl = fileSystemName + "@" + accountName;
+    URI defaultUri = null;
+
+    try {
+      defaultUri = new URI(abfsScheme, abfsUrl, null, null, null);
+    } catch (Exception ex) {
+      throw new AssertionError(ex);
+    }
+
+    String testUrl = defaultUri.toString();
+    abfsConfig.set(FS_DEFAULT_NAME_KEY, defaultUri.toString());
```

Comment:
We can inherit `AbstractAbfsIntegrationTest`, which would give us all the boilerplate for running the  test. What we can do, is get the configuration form the getFileSystem().getConf(); then, make changes to the config, and then return back filesystem created with FileSystem.newInstance()..  What you feel?